frontend_lib/src/appContentFactory.js

```diff
@@ -0,0 +1,380 @@
+        msg: msg,
+        type: type || 'warning',
+        delay: delay || undefined
+      }
+    })
+
+    // INFO - CH - 2019-01-08 - event called by OpenContentApp to open the show the app if it is already rendered
+    appContentCustomEventHandlerShowApp = (newContent, content, setState, buildBreadcrumbs) => {
+      if (newContent.content_id !== content.content_id) {
+        GLOBAL_dispatchEvent({ type: CUSTOM_EVENT.RELOAD_CONTENT(content.content_type), data: newContent })
+        return
+      }
+      setState({ isVisible: true })
+      buildBreadcrumbs()
+    }
+
+    // INFO - CH - 2019-01-08 - event called by OpenContentApp in case of opening another app feature
+    appContentCustomEventHandlerHideApp = setState => {
+      tinymce.remove('#wysiwygTimelineComment')
+      setState({
+        isVisible: false,
+        timelineWysiwyg: false
+      })
+    }
+
+    // CH - 2019-31-12 - This event is used to send a new content_id that will trigger data reload through componentDidUpdate
+    appContentCustomEventHandlerReloadContent = (newContent, setState, appSlug) => {
+      tinymce.remove('#wysiwygTimelineComment')
+
+      const previouslyUnsavedComment = localStorage.getItem(
+        generateLocalStorageContentId(newContent.workspace_id, newContent.content_id, appSlug, 'comment')
+      )
+
+      setState(prev => ({
+        content: { ...prev.content, ...newContent },
+        isVisible: true,
+        timelineWysiwyg: false,
+        newComment: prev.content.content_id === newContent.content_id
+          ? prev.newComment
+          : (previouslyUnsavedComment || '')
+      }))
+    }
+
+    // CH - 2019-31-12 - This event is used to reload all app data. It's not supposed to handle content id change
+    appContentCustomEventHandlerReloadAppFeatureData = async (loadContent, loadTimeline, buildBreadcrumbs) => {
+      await loadContent()
+      loadTimeline()
+      buildBreadcrumbs()
+    }
+
+    // INFO - 2019-01-09 - if param isTimelineWysiwyg is false, param changeNewCommentHandler isn't required
+    appContentCustomEventHandlerAllAppChangeLanguage = (newLang, setState, i18n, isTimelineWysiwyg, changeNewCommentHandler = null) => {
+      if (isTimelineWysiwyg) {
+        tinymce.remove('#wysiwygTimelineComment')
+        wysiwyg('#wysiwygTimelineComment', newLang, changeNewCommentHandler)
+      }
+
+      setState(prev => ({
+        loggedUser: {
+          ...prev.loggedUser,
+          lang: newLang
+        }
+      }))
+      i18n.changeLanguage(newLang)
+    }
+
+    appContentChangeTitle = async (content, newTitle, appSlug, propertiesToAddToBody = {}) => {
+      this.checkApiUrl()
+
+      if (content.label === newTitle) return
+
+      // INFO - CH - 2019-01-03 - Check the -s added to the app slug. This is and should stay consistent with app features
+      const response = await handleFetchResult(
+        await fetch(`${this.apiUrl}/workspaces/${content.workspace_id}/${appSlug}s/${content.content_id}`, {
```

Comment:
For a better consistency, maybe should we move `frontend_lib` api calls in a `src/action.async.js` file like in apps ?

Reply:
I agree. I'll do the changes